src/onboarding/registration/NameAndPicture.tsx

```diff
@@ -35,9 +35,22 @@ import { saveProfilePicture } from 'src/utils/image'
 import Logger from 'src/utils/Logger'
 import { useAsyncKomenciReadiness } from 'src/verify/hooks'
 import { Statsig } from 'statsig-react-native'
+import { ADJECTIVES, NOUNS } from './constants'
 
 type Props = StackScreenProps<StackParamList, Screens.NameAndPicture>
 
+//Exported for testing purposes only
+export const generateUsername = (): string => {
+  const adjective = ADJECTIVES[Math.floor(Math.random() * ADJECTIVES.length)]
+  const noun = NOUNS[Math.floor(Math.random() * NOUNS.length)]
+  const username = `${adjective} ${noun}`
+  const forbiddenUsernames: string[] = [] //TODO: Obtain username blacklist from Firebase Remote Config
+  if (forbiddenUsernames.includes(username)) {
```

Comment:
one more thing-- the worst case runtime for this is unbounded. seems simpler to just filter the word lists for anything on the blacklist before using them.